File: Lab_2/output/input1.py
@dataclass
class PersonData:
    personId: float
    personName: str
    contactEmail: str
    yearsOld: float
    accountActive: bool

@dataclass
class ItemForSale:
    itemId: float
    itemName: str
    costValue: float

def makeNewPerson(identificationNumber: float, givenName: str, electronicMail: str, ageInYears: float) -> PersonData:
    return {
    personId: identificationNumber,
    personName: givenName,
    contactEmail: electronicMail,
    yearsOld: ageInYears,
    accountActive: True
    }

def computePriceReduction(baseCost: float, reductionPercent: float) -> float:
    return baseCost - baseCost * reductionPercent / 100

checkIfLegalAge = lambda ageValue: ageValue >= 18  # type: (float) -> bool

raiseCostByTen = lambda originalCost: originalCost + 10  # type: (float) -> float

def displayPersonName(personRecord: PersonData) -> None:
    print(personRecord.personName)

File: Lab_2/input/input1.ts
interface PersonData {
  personId: number;
  personName: string;
  contactEmail: string;
  yearsOld: number;
  accountActive: boolean;
}

interface ItemForSale {
  itemId: number;
  itemName: string;
  costValue: number;
}

function makeNewPerson(
  identificationNumber: number,
  givenName: string,
  electronicMail: string,
  ageInYears: number
): PersonData {
  return {
    personId: identificationNumber,
    personName: givenName,
    contactEmail: electronicMail,
    yearsOld: ageInYears,
    accountActive: true
  };
}

function computePriceReduction(
  baseCost: number,
  reductionPercent: number
): number {
  return baseCost - baseCost * reductionPercent / 100;
}

const checkIfLegalAge = (ageValue: number): boolean => ageValue >= 18;

const raiseCostByTen = (originalCost: number): number => originalCost + 10;

const displayPersonName = (personRecord: PersonData): void => {
  console.log(personRecord.personName);
};